system-tests/lib/infra/kubernetes.go

```diff
@@ -0,0 +1,187 @@
+package infra
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/framework/components/clnode"
+	"github.com/smartcontractkit/chainlink-testing-framework/framework/components/jd"
+	ns "github.com/smartcontractkit/chainlink-testing-framework/framework/components/simple_node_set"
+)
+
+const (
+	K8sConfigsDir = "k8s-configs"
+
+	// Kubernetes service naming patterns
+	NodeBootstrapNameFormat = "%s-bt-%d" // e.g., workflow-bt-0
+	NodePluginNameFormat    = "%s-%d"    // e.g., workflow-0
+
+	// Job Distributor service names
+	JDGRPCServiceName    = "job-distributor:80"
+	JDNodeRPCServiceName = "job-distributor-noderpc-lb:80"
+)
+
+// KubernetesInput contains configuration for Kubernetes-based infrastructure
+type KubernetesInput struct {
+	Namespace string `toml:"namespace" validate:"required"`
+	// absolute path to the folder with Kubernetes CRE configs
+	FolderLocation  string `toml:"folder_location"`
+	ExternalDomain  string `toml:"external_domain"`
+	ExternalPort    int    `toml:"external_port"`
+	LabelSelector   string `toml:"label_selector"`
+	NodeAPIUser     string `toml:"node_api_user"`
+	NodeAPIPassword string `toml:"node_api_password"`
+	// Use TLS for external JD GRPC connections (default: true for port 443, false otherwise)
+	UseTLSForJD *bool `toml:"use_tls_for_jd"`
+	// Kubernetes provider type (aws, kind, etc.) - optional
+	Provider string `toml:"provider" validate:"omitempty,oneof=aws kind"`
+	// Required for cost attribution in AWS
+	TeamInput *Team `toml:"team_input" validate:"required_if=Provider aws"`
+}
+
+// GetKubernetesClient creates a Kubernetes client from the default kubeconfig or in-cluster config
+func GetKubernetesClient() (*kubernetes.Clientset, error) {
+	// Try to use in-cluster config first (when running inside a pod)
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		// Fall back to kubeconfig
+		loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+		configOverrides := &clientcmd.ConfigOverrides{}
+		kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+		config, err = kubeConfig.ClientConfig()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get kubernetes config: %w", err)
+		}
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kubernetes clientset: %w", err)
+	}
+
+	return clientset, nil
+}
+
+// GenerateKubernetesJDOutput generates JD service URLs for Kubernetes-deployed Job Distributor
+func GenerateKubernetesJDOutput(infraInput *Provider, lggr zerolog.Logger) *jd.Output {
+	externalDomain := ""
+
+	if infraInput.Kubernetes != nil {
```

Comment:
why not just return err if `infraInput.Kubernetes` is `nil` and never check it again?